localstack-core/localstack/services/stepfunctions/provider.py

```diff
@@ -927,18 +1149,50 @@ def delete_state_machine(
                 state_machines.pop(version_arn, None)
         return DeleteStateMachineOutput()
 
+    def delete_state_machine_alias(
+        self, context: RequestContext, state_machine_alias_arn: Arn, **kwargs
+    ) -> DeleteStateMachineAliasOutput:
+        self._validate_state_machine_alias_arn(state_machine_alias_arn=state_machine_alias_arn)
+        store = self.get_store(context=context)
+        aliases = store.aliases
+        if (alias := aliases.pop(state_machine_alias_arn, None)) is not None:
+            state_machines = store.state_machines
+            for routing_configuration in alias.get_routing_configuration_list():
+                state_machine_version_arn = routing_configuration["stateMachineVersionArn"]
+                if (
+                    state_machine_version := state_machines.get(state_machine_version_arn)
+                ) is None or not isinstance(state_machine_version, StateMachineVersion):
+                    continue
+                if (
+                    state_machine_revision := state_machines.get(state_machine_version.source_arn)
+                ) is None or not isinstance(state_machine_revision, StateMachineRevision):
+                    continue
+                state_machine_revision.aliases.remove(alias)
```

Comment:
When the`StateMachineRevision` is retrieved from the store, this `state_machine_revision.aliases.remove(alias)` will raise an exception. I noticed this being raised in `test_base_lifecycle_create_update_describe`.

Perhaps rather use the idempotent `discard` which wont raise an exception when trying to delete a non-existent key from a set:
```suggestion
                if (
                    state_machine_revision := state_machines.get(state_machine_version.source_arn)
                ) is None or not isinstance(state_machine_revision, StateMachineRevision):
                    continue
                state_machine_revision.aliases.discard(alias)
```